src/test/java/seedu/duke/template/LcTemplateTest.java

```diff
@@ -2,14 +2,109 @@
 
 import org.junit.jupiter.api.Test;
 import seedu.duke.DukeException;
+import seedu.duke.component.Capacitor;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class LcTemplateTest {
+    public static final double DELTA = 1e-15;
 
     @Test
     void calcImpedance_zeroCapacitance_expectException() {
         LcTemplate t = new LcTemplate();
         assertThrows(DukeException.class, t::calcImpedance);
     }
+
+    @Test
+    void calcImpedance_returnsDouble() throws DukeException {
+        LcTemplate t = new LcTemplate(1,1,1);
+        double inductance = 1 * Math.pow(10, -6);
+        double capacitance = 1 * Math.pow(10, -6);
+        double impedance = t.ANGULAR_FREQUENCY * Math.abs(inductance - (1 / capacitance));
+        assertEquals(impedance, t.calcImpedance(), DELTA);
+    }
+
+    @Test
+    void getInductor_returnsInductor() {
```

Comment:
Consider adding edge cases? Like 0, or negative values, to make the code more robust - similar for all the methods.

Reply:
Right, I'll figure that out. Thanks Wira :)